src/jdk.jfr/share/classes/jdk/jfr/internal/tool/PrettyWriter.java

```diff
@@ -437,13 +437,16 @@ void printObject(RecordedObject object, long arraySize) {
     private void printClassLoader(RecordedClassLoader cl, String postFix) {
         // Purposely not printing class loader name to avoid cluttered output
         RecordedClass clazz = cl.getType();
-        print(clazz == null ? "null" : clazz.getName());
         if (clazz != null) {
+            print(clazz.getName());
             print(" (");
             print("id = ");
             print(String.valueOf(cl.getId()));
-            println(")");
+            print(")");
+        }else{
+            print("null");
         }
+        println(postFix);
```

Comment:
Categorical postFix now possible by changing 445 from println(")"); to print(")");?

Reply:
yes exactly. I think this is how this method was supposed to work. I compared it with similar methods like printClass().

If there is a postFix, its always printed, and it adds a new line in this case. The new line was conditional before (which caused the bug in the first place).